packages/picasso.js/src/core/chart-components/point/point.js

```diff
@@ -127,15 +127,22 @@ function getType(s) {
   return [type, props];
 }
 
-function createDisplayPoints(dataPoints, { width, height }, pointSize, shapeFn) {
+function createDisplayPoints(dataPoints, { width, height }, pointSize, shapeFn, errorShape) {
   return dataPoints
     .filter((p) => p.show !== false && !isNaN(p.x + p.y))
     .map((p) => {
       let s = p;
       let size = PX_RX.test(p.size) ? parseInt(p.size, 10) : pointSize.min + s.size * (pointSize.max - pointSize.min);
       if (notNumber(size)) {
-        s = DEFAULT_ERROR_SETTINGS.errorShape;
-        size = pointSize.min + s.size * (pointSize.max - pointSize.min);
+        s = {
+          fill: p.fill,
+          stroke: p.stroke,
+          strokeWidth: p.strokeWidth,
+          strokeDasharray: p.strokeDasharray,
+          opacity: p.opacity,
+          ...(errorShape || DEFAULT_ERROR_SETTINGS.errorShape),
+        };
```

Comment:
I have seen this pattern before and don't like it since it's a risk of writing the wrong key…
Can we do something like this – or is there an ES6+ way of doing this?
```
const pickProps = (propsArr, obj) => Object.fromEntries(Object.entries(p).filter((entry) => propsArr.includes(entry[0]) ? entry : false));
s = {
  ...pickProps(['fill', 'stroke', 'strokeWidth', 'strokeDasharray', 'opacity'], p),
  ...(errorShape || DEFAULT_ERROR_SETTINGS.errorShape),
};

```

Reply:
I don't want to make it complicated in this case!